src/MainOpt.cxx

```diff
@@ -28,7 +31,7 @@ using std::vector;
 MainOpt::MainOpt() {
   hostname.resize(128);
   gethostname(hostname.data(), hostname.size());
-  if (hostname.back() != 0) {
+  if (hostname.back()) {
```

Comment:
I think this makes it less readable. The type of `hostname.back()` is neither `bool` nor a pointer.